### ISSUE TITLE
Search results modal focus flow is faulty

### ISSUE THREAD

**Issue by DanielCouzens** [epimorphics/ppd-explorer#216]
- [x] The user does a search and then tabs through the results screen to the share view button. The user starts with the link in focus. The user can then tab to 'tweet' and then tab to something behind the modal when the user would expect to tab to the 'ok' button. 

- [x] Tabbing further takes me onto items behind the modal but the desired behaviour should be to remain focussed on the modal elements.

https://docs.google.com/spreadsheets/d/1YS3N5ps9pJGFOxBDi9C2DwA37WQPIVax/edit?gid=154688571#gid=154688571&range=A5

**Comment by DanielCouzens** [comment]
Created a Draft PR https://github.com/epimorphics/ppd-explorer/pull/223

**Comment by DanielCouzens** [comment]
In review PR https://github.com/epimorphics/ppd-explorer/pull/223

**Comment by DanielCouzens** [comment]
I have closed Pr https://github.com/epimorphics/ppd-explorer/pull/223 and combined issue 216 and 218 into one PR

please see PR https://github.com/epimorphics/ppd-explorer/pull/224

**Comment by DanielCouzens** [comment]
PR https://github.com/epimorphics/ppd-explorer/pull/224 also fixes the following issues from the report

- [x] [2.1.2: no Keyboard trap](https://docs.google.com/spreadsheets/d/1YS3N5ps9pJGFOxBDi9C2DwA37WQPIVax/edit?gid=154688571#gid=154688571&range=A11)
- [x] [2.4.11: Focus Not Obscured (Minimum)](https://docs.google.com/spreadsheets/d/1YS3N5ps9pJGFOxBDi9C2DwA37WQPIVax/edit?gid=154688571#gid=154688571&range=A13)
- [x] [top links aren't accessible easily via keyboard ](https://docs.google.com/spreadsheets/d/1YS3N5ps9pJGFOxBDi9C2DwA37WQPIVax/edit?gid=154688571#gid=154688571&range=A16)

**Comment by DanielCouzens** [comment]
@tomguilbert this is ready to check on the dev instance https://hmlr-dev-pres.epimorphics.net/app/ppd/

**Comment by tomguilbert** [comment]
Looks good thanks Dan!
